sycl/test/plugins/enqueue-arg-order.cpp

```diff
@@ -0,0 +1,261 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+// RUN: env SYCL_PI_TRACE=2 %CPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+// XFAIL: *
+
+/*
+  Manual
+    clang++ -fsycl -o eao.bin enqueue-arg-order.cpp
+    SYCL_PI_TRACE=2 ./eao.bin
+
+    clang++ --driver-mode=g++ -fsycl
+  -fsycl-targets=nvptx64-nvidia-cuda-sycldevice -o eao.bin enqueue-arg-order.cpp
+    SYCL_PI_TRACE=2 SYCL_BE=PI_CUDA ./eao.bin
+
+    llvm-lit --param SYCL_BE=PI_CUDA -v enqueue-arg-order.cpp
+*/
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+constexpr long width = 16;
+constexpr long height = 5;
+constexpr long total = width * height;
+
+void remind() {
+  /*
+    https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clEnqueueReadBufferRect.html
+
+    buffer_origin defines the (x, y, z) offset in the memory region associated
+    with buffer. For a 2D rectangle region, the z value given by
+    buffer_origin[2] should be 0. The offset in bytes is computed as
+    buffer_origin[2] × buffer_slice_pitch + buffer_origin[1] × buffer_row_pitch
+    + buffer_origin[0].
+
+    region defines the (width in bytes, height in rows, depth in slices) of the
+    2D or 3D rectangle being read or written. For a 2D rectangle copy, the depth
+    value given by region[2] should be 1. The values in region cannot be 0.
+
+
+    buffer_row_pitch is the length of each row in bytes to be used for the
+    memory region associated with buffer. If buffer_row_pitch is 0,
+    buffer_row_pitch is computed as region[0].
+
+    buffer_slice_pitch is the length of each 2D slice in bytes to be used for
+    the memory region associated with buffer. If buffer_slice_pitch is 0,
+    buffer_slice_pitch is computed as region[1] × buffer_row_pitch.
+  */
+  std::cout << "For BUFFERS" << std::endl;
+  std::cout << "         Region SHOULD be : " << width * sizeof(float) << "/"
+            << height << "/" << 1 << std::endl; // 64/5/1
+  std::cout << "  RowPitch SHOULD be 0 or : " << width * sizeof(float)
+            << std::endl; // 0 or 64
+  std::cout << "SlicePitch SHOULD be 0 or : " << width * sizeof(float) * height
+            << std::endl
+            << std::endl; // 0 or 320
+
+  // NOTE: presently we see 20/16/1 for Region and 20 for row pitch.  both
```

Comment:
Let's commit this test with the actual fix for this?

Reply:
agreed. Removed the test.